onnxruntime/core/optimizer/graph_transformer_mgr.h

```diff
@@ -4,31 +4,63 @@
 #pragma once
 
 #include "core/optimizer/graph_transformer.h"
-
+#include "core/optimizer/graph_transformer_factory.h"
+using namespace ::onnxruntime::common;
 namespace onnxruntime {
-// Manages a list of graph transformers. It is initialized with a list of graph
+
+// Manages graph transformation for a inference session. It is initialized with a list of pre-defined graph
 // transformers. Each inference session can further register additional ones.
 class GraphTransformerManager {
  public:
-  explicit GraphTransformerManager(unsigned steps) noexcept : steps_(steps) {
-    // TODO: Register default transformers.
+  explicit GraphTransformerManager(unsigned steps, unsigned int level) : steps_(steps), factory_{std::make_unique<GraphTransformerFactory>()} {
+    ORT_ENFORCE(ValidateOptimizationLevel(level).IsOK(),
+                "Unsupported optimization level specified");
+
+    optimizations_enabled_ |= TransformerLevel::Default_ProviderSpecific;
+    if (level == 1) {
+      optimizations_enabled_ |= TransformerLevel::Optional_L1;
+    } else if (level == 2) {
+      optimizations_enabled_ |= TransformerLevel::Optional_L1 | TransformerLevel::Optional_L2;
+    }    
   }
 
-  // Register a graph transformer.
-  common::Status Register(std::unique_ptr<GraphTransformer> transformer) {
-    transformers_.push_back(std::move(transformer));
-    return common::Status::OK();
+  // Registers a graph transformer.
+  Status Register(std::unique_ptr<GraphTransformer> transformer) {
+      return factory_->Register(std::move(transformer));
   }
 
-  // Apply the list of graph transformers registered on the specified graph
+  // Apply the list of graph transformers registered for the specified level on the specified graph
   // up to the given number of steps.
-  common::Status ApplyAll(Graph& graph) const;
+  Status ApplyTransformations(Graph& graph, const std::vector<TransformerLevel>& levels) const;
+  
+  Status ApplyTransformations(Graph& graph, const std::vector<std::string>& transformer_ids) const;
+
+  Status ApplyTransformations(Graph& graph, const GraphTransformer* transformer) const;
 
  private:
-  GraphTransformerManager() = default;
+  using TransformersList = std::vector<GraphTransformer*>;  
+
```

Comment:
It feels like this alias just makes the code harder to read. I understand a 'using' alias to shorten really long names or abstract away something complicated, but here we don't see to be gaining much benefit from the obfuscation of the types involved.